### ISSUE TITLE
Running on Google Colab with GPU Dependency Issues: invalid device ordinal

### ISSUE THREAD

**Issue by zhichucode** [microsoft/visual-chatgpt#27]
https://colab.research.google.com/drive/1inhlmcp9GB611bUpE6GKjyGguw3Ub9Or?usp=sharing

I don't have a powerful GPU, so I'm attempting to run this repository on Google Colab. I was able to successfully install all of the necessary dependencies, but I encountered an error when attempting to run `visual_chatgpt.py`:

```

RuntimeError: CUDA error: invalid device ordinal
```

Has anyone been successful in running this repository on Google Colab?

**Comment by K-tang-mkv** [comment]
@zhichucode 
maybe there was not enough gpu on your Colab, look at this, to many cuda device needed
![Screen Shot 2023-03-10 at 11 09 43 AM](https://user-images.githubusercontent.com/59059785/224213499-292a0186-739f-43f0-ab92-d2d38e86bddb.png)

**Comment by zhichucode** [comment]
Can I just use one of these features instead of all? I don't have much CPUs.

**Comment by rupeshs** [comment]
I have added colab support see this https://github.com/rupeshs/visual-chatgpt/tree/add-colab-support

**Comment by powderblock** [comment]
I had to change all instances of cuda:n where n is a number to cuda:0 since I only have one GPU. After changing it to cuda:0 the script ran for me. (Ran out of GPU memory but still, at least it started up!)

**Comment by rupeshs** [comment]
Yes it will work

**Comment by RalphX1** [comment]
> I had to change all instances of cuda:n where n is a number to cuda:0 since I only have one GPU. After changing it to cuda:0 the script ran for me. (Ran out of GPU memory but still, at least it started up!)

same

**Comment by K-tang-mkv** [comment]
@powderblock yeah, only using one CUDA is fine, but GPU memory is clearly not enough for those models. To save your GPU memory, you can properly modify `self.tools` with fewer visual foundation models. 
![Screen Shot 2023-03-11 at 7 48 26 AM](https://user-images.githubusercontent.com/59059785/224449281-d478d81f-eea5-4e04-a0cd-b06a3f94729d.png)

**Comment by randerzander** [comment]
> @powderblock yeah, only using one CUDA is fine, but GPU memory is clearly not enough for those models. To save your GPU memory, you can properly modify `self.tools` with fewer visual foundation models. ![Screen Shot 2023-03-11 at 7 48 26 AM](https://user-images.githubusercontent.com/59059785/224449281-d478d81f-eea5-4e04-a0cd-b06a3f94729d.png)

Any one have an idea of GPU memory usage for each of these tools?

**Comment by rupeshs** [comment]
@randerzander yes,it is in the readme https://github.com/microsoft/visual-chatgpt#gpu-memory-usage

**Comment by K-tang-mkv** [comment]
@randerzander Under google Colab standard GPU, you can't choose too many models to run it because of its insufficient GPU memory. I already implement the Google Colab version and I only choose two models `T2I` and `ImageCaption`, everything is fine, you can try it on my Colab notebook, checkout this [https://github.com/K-tang-mkv/visual-chatgpt-googlecolab]()
